src/lms-ds-loader/edfi_lms_ds_loader/migrator.py

```diff
@@ -0,0 +1,124 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import logging
+from os import path
+from typing import Any, Callable, List
+
+from sqlalchemy.engine.base import Engine
+from sqlalchemy.orm import sessionmaker, session as sa_session
+from sqlalchemy.exc import ProgrammingError
+from sqlparse import split
+
+logger = logging.getLogger(__name__)
+
+MIGRATION_SCRIPTS = [
+    "initialize_lms_database",
+    "create_user_tables"
+]
+
+
+def _get_script_path(engine: Engine, script_name: str) -> str:
+    script_dir = path.join(path.dirname(__file__), "scripts", engine.name)
+    return path.join(script_dir, script_name)
+
+
+def _read_statements_from_file(full_path: str) -> List[str]:
+    raw_sql: str
+    with open(full_path) as f:
+        raw_sql = f.read()
+
+    return split(raw_sql)
+
+
+def _execute_statements(engine: Engine, statements: List[str]):

```

Comment:
TODO: refactor this function or get rid of it entirely in favor of `_execute_transaction`.

Reply:
done